From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Spottedleaf <Spottedleaf@users.noreply.github.com>
Date: Mon, 2 Dec 2019 05:15:55 -0800
Subject: [PATCH] Optimise ActivationRange#activateEntities(World)

By using our Chunk#forEachEntity method, the time trimmed was
approximately 2-5x in my testing.

diff --git a/src/main/java/org/spigotmc/ActivationRange.java b/src/main/java/org/spigotmc/ActivationRange.java
index b994ee98e036aa24f25ebb9654d1f74c847570a0..91b6555ed15e73a98fd902aadf9dd1e72ed86061 100644
--- a/src/main/java/org/spigotmc/ActivationRange.java
+++ b/src/main/java/org/spigotmc/ActivationRange.java
@@ -162,24 +162,15 @@ public class ActivationRange
      */
     private static void activateChunkEntities(Chunk chunk)
     {
-        for ( List<Entity> slice : chunk.entitySlices )
-        {
-            for ( Entity entity : (Collection<Entity>) slice )
-            {
+        chunk.forEachEntity((entity) -> { /* Concrete - optimise this call */
                 if ( MinecraftServer.currentTick > entity.activatedTick )
                 {
-                    if ( entity.defaultActivationState )
-                    {
-                        entity.activatedTick = MinecraftServer.currentTick;
-                        continue;
-                    }
-                    if ( entity.activationType.boundingBox.c( entity.getBoundingBox() ) )
+                    if ( entity.defaultActivationState || entity.activationType.boundingBox.intersects(entity.getBoundingBox())) /* Concrete - optimise this call */
                     {
                         entity.activatedTick = MinecraftServer.currentTick;
-                    }
+                    } /* Concrete - optimise this call */
                 }
-            }
-        }
+        }); /* Concrete - optimise this call */
     }
 
     /**
-- 
2.24.0.windows.2

